src/Middleware/AuthenticationMiddleware.php

```diff
@@ -146,7 +146,9 @@ protected function getRedirectUrl($target, ServerRequestInterface $request)
             return $target;
         }
 
-        $query = urlencode($param) . '=' . urlencode($request->getUri());
+        $uri = $request->getUri();
+        $uri = $uri->withPath($uri->base . $uri->getPath());
```

Comment:
Should we make sure that the uri has a `base` property? Something like `if (property_exists($uri, 'base')) {` should work. My concern is that we could get vanilla Uri objects and hit runtime errors.

Reply:
The uri we deal with here is the one parsed by the request, which I believe will always have the base set, though it may be null. Unless we expect this may be used with something other than Cake's ServerRequestFactory?

Reply:
Not sure if you're waiting on me to update the PR based on this...

Reply:
We tried to make this middleware usable in non-cakephp projects. Doing so requires sticking to the PSR7 interface methods and attributes and only conditionally using our extensions if they are available.